pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -111,15 +117,15 @@ func (n *NetworkPolicyController) filterPerNamespaceRuleACNPsByNSLabels(nsLabels
 
 	peerNamespacesSelectorExists := func(peers []crdv1beta1.NetworkPolicyPeer) bool {
 		for _, peer := range peers {
-			if peer.Namespaces != nil && peer.Namespaces.Match == crdv1beta1.NamespaceMatchSelf {
+			if peer.Namespaces != nil {
 				return true
 			}
 		}
 		return false
 	}
 
 	affectedPolicies := sets.New[string]()
-	objs, _ := n.acnpInformer.Informer().GetIndexer().ByIndex(perNamespaceRuleIndex, HasPerNamespaceRule)
+	objs, _ := n.acnpInformer.Informer().GetIndexer().ByIndex(perNamespaceRuleIndex, hasSuchRule)
```

Comment:
Not related to this change, but the line 134 to 137 can be merged to `return namespaceLabelMatches(cnp.Spec.AppliedTo)`.

Reply:
Done